sympy/core/mul.py

```diff
@@ -1255,19 +1255,20 @@ def _eval_is_integer(self):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
-            if d is S.One:
+            if d == S.One:
                 return True
             elif d == S(2):
                 return n.is_even
-        # if d is even -- 0 or not -- the
-        # result is not an integer
         if n.is_odd and d.is_even:
             return False
+        if n == d:
+            return True
+        if is_rational is None:
```

Comment:
check coverage -- I don't think you can ever get anything but None from these returns: if the denominator was a literal 1 or 2 and `is_rational` is None then that means we don't know if it is an integer and we won't know if it is even.

```python
    def _eval_is_integer(self):
        from sympy import fraction

        is_rational = self._eval_is_rational()
        if is_rational is False:
            return False

        n, d = fraction(self)
        if is_rational:
            if d is S.One:
                return True
            if d.is_even
                if d.is_prime:  # 2
                    return n.is_even
                if n.is_odd:
                    return False  # true whether d is 0 or not
            if n == d:  # infrequent case
                return True
```

Reply:
> # [Codecov](https://codecov.io/gh/sympy/sympy/pull/19182?src=pr&el=h1) Report
> 
> > Merging [#19182](https://codecov.io/gh/sympy/sympy/pull/19182?src=pr&el=desc) into [master](https://codecov.io/gh/sympy/sympy/commit/8c94b6428b4d5783077cea1afe9fcbad2be3ec91&el=desc) will **increase** coverage by `0.011%`.
> > The diff coverage is `100.000%`.
> 
> ```diff
> @@              Coverage Diff              @@
> ##            master    #19182       +/-   ##
> =============================================
> + Coverage   75.666%   75.677%   +0.011%     
> =============================================
>   Files          651       651               
>   Lines       169336    169339        +3     
>   Branches     39970     39972        +2     
> =============================================
> + Hits        128131    128152       +21     
> + Misses       35605     35592       -13     
> + Partials      5600      5595        -5     
> ```

Reply:
yes...but those lines?

Reply:
I see that it says 100% -- I would have to see the test that passes through those lines to educate my imagination. (Which we know can be short-sighted.)

Reply:
I'm writing up tests for them, I'll post it here before pushing 😃

Reply:
> before pushing

good idea...I'll stop tests so you can get the `n==d` condition fixed, too.

Reply:
all lines covered ? https://github.com/sympy/sympy/pull/19182#issuecomment-619165555

> I would have to see the test that passes through those lines to educate my imagination.